sos/report/plugins/postfix.py

```diff
@@ -52,6 +54,38 @@ def forbidden_ssl_keys_files(self):
         finally:
             return fp
 
+    def forbidden_password_files(self):
+        forbid_attributes = (
+            "lmtp_sasl_password_maps",
+            "smtp_sasl_password_maps",
+            "postscreen_dnsbl_reply_map",
+            "smtp_sasl_auth_cache_name",
+        )
+        fp = []
+        prefix = 'hash:'
+        option_format = re.compile(r"^(.*)=(.*)")
+        try:
+            with open(self.path_join('/etc/postfix/main.cf'), 'r') as cffile:
+                for line in cffile.readlines():
+                    # ignore comment and check option format
+                    line = re.sub('#.*', '', line)
+                    option = option_format.match(line)
+                    if option is None:
+                        continue
+
+                    # sieving
+                    if option.group(1).strip() in forbid_attributes:
+                        filepath = option.group(2).strip()
+                        # ignore no filepath
+                        if len(filepath) == 0:
+                            continue
+                        # remove prefix
+                        if filepath.startswith(prefix):
+                            filepath = filepath[len(prefix):]
+                        fp.append(filepath)
+        finally:
+            return fp
```

Comment:
A return in a `finally` block will swallow raised exceptions. I think we should instead have an exception handler (log it for traceability) and have the return outside the try/except block.